docs/spec-files/general_messaging_constructs.md

```diff
@@ -10,6 +10,14 @@
 * Errors
 * Tracing / troubleshooting
 
+## Goal Codes
+
+Goal codes are used to coordinate the purpose of an interaction. Some protocols are generic enough to be used for different purposes, and goal codes are a mechanism to communicate the purpose of an interaction in that situation. They are used in Out of Band messages for that reason.
```

Comment:
When I re-read through this I keep getting caught on the purpose of these. I think it's because we're coupling the semantics of human meaningful interactions to a particular identifier so that we can infer how to process the interaction at a technical level and that seems off to me when we've already essentially got that in the protocol identifiers. Essentially I remain confused on when I should be differentiating processing based on a goal code rather than based on the protocol in use.

From this description it sounds like goal codes are being used as an encapsulation mechanism to describe a [coprotocol](https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0478-coprotocols) and I'm wondering why a coprotocol URI couldn't be defined and to drop this.